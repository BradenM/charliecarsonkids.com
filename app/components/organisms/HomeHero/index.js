/**
 *
 * HomeHero
 *
 * Organism
 */

import Hero from 'molecules/Hero';
import React from 'react';
import Container from 'react-bootstrap/Container';
import styled from 'styled-components';
import { ImageURL } from 'utils/getStatic';

const HeroContainer = styled(Container)`
  width: 100%;
  min-height: 70vh;
  .hero {
    box-shadow: 0px 0px 20px 0px RGBA(39, 40, 43, 0.15);
  }
`;

const HeroProps = {
  text: 'new arrivals',
  buttonText: 'Shop Now',
  imageSrc: ImageURL('hero'),
  onButtonClick: () => {},
};

const HomeHero = () => (
  <HeroContainer>
    <Hero {...HeroProps} />
  </HeroContainer>
);

export default HomeHero;
