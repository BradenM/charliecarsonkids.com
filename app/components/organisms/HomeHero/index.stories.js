/**
 *
 * Stories for HomeHero
 *
 * Organism
 */

import React from 'react';
import HomeHero from '.';

export default {
  title: 'Design|Organisms/HomeHero',
  component: HomeHero,
  parameters: {
    centered: { disable: true },
  },
};

export const defaultStory = () => <HomeHero />;
