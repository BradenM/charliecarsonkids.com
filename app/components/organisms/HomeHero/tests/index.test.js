/**
 *
 * Tests for HomeHero
 *
 * Organism
 */

import React from 'react';
import { render } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions
import HomeHero from '../index';

describe('<HomeHero />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<HomeHero />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<HomeHero />);
    expect(firstChild).toMatchSnapshot();
  });
});
