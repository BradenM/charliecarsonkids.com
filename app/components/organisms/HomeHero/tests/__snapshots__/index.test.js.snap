// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeHero /> Should render and match the snapshot 1`] = `
<div
  class="HomeHero__HeroContainer-sc-18ez42y-0 hdBiEm container"
>
  <div
    class="hero Hero__StyledJumbotron-sc-1ele3ck-0 nwfXD jumbotron"
  >
    <div
      class="h-100 container"
    >
      <div
        class="row"
      >
        <div
          class="pb-3 my-auto col-6"
        >
          <h1
            class="Title-sc-3abvrr-0 mb-0 font-weight-bold Hero__HeroTitle-sc-1ele3ck-1 display-1 mb-0 font-weight-bold bqYlOi"
            color="#fefefe"
            size="1"
          >
            NEW ARRIVALS
          </h1>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="py-3 col-3"
        >
          <button
            class="btn btn-outline-light btn-lg"
            type="button"
          >
            Shop Now
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
